### PR TITLE
Fix/labels

### PR DESCRIPTION
**Pull request by loicmathieu**

### CODE REVIEW

File: ui/src/components/layout/Labels.vue [hunk 1]
```diff
@@ -29,6 +29,23 @@
                 default: true
             }
         },
+        // this is needed as flows uses a Map and Execution a List of Labels.
+        // if we align both of them this can be removed
+        mounted() {
+            if (Array.isArray(this.labels)) {
+                const labelMap = {};
+                this.labels.forEach(label => labelMap[label.key] = label.value)
+                this.labelMap = labelMap;
```
Comment (brian-mulier-p) [inline ui/src/components/layout/Labels.vue:38]: this.labelMap = Object.fromEntries(this.labels.map(label => [label.key, label.value]))

File: ui/src/components/layout/Labels.vue [hunk 2]
```diff
@@ -29,6 +29,23 @@
...
+            } else {
+                this.labelMap = this.labels;
+            }
+            console.log(this.labelMap)
```
Comment (brian-mulier-p) [inline ui/src/components/layout/Labels.vue:42]:
```suggestion
```

**Review by brian-mulier-p — CHANGES_REQUESTED** [review]
Some slight adjustments but looks good otherwise

File: core/src/main/java/io/kestra/core/serializers/ListOrMapOfLabelDeserializer.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        else if (p.hasToken(JsonToken.START_ARRAY)) {
+            // deserialize as list
+            List<Map<String, String>> ret = ctxt.readValue(p, List.class);
+            return ret.stream().map(map -> new Label(map.get("key"), map.get("value"))).toList();
+        }
+        else if (p.hasToken(JsonToken.START_OBJECT)) {
+            // deserialize as map
+            Map<String, String> ret = ctxt.readValue(p, Map.class);
+            return ret.entrySet().stream().map(entry -> new Label(entry.getKey(), entry.getValue())).toList();
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply core/src/main/java/io/kestra/core/serializers/ListOrMapOfLabelDeserializer.java:33]: Thanks, I miss this one :(

File: ui/src/components/layout/Labels.vue (hunk 1 above)
Comment (loicmathieu) [reply ui/src/components/layout/Labels.vue:38]: Well, maybe but as I don't want to re-test the UI I'll keep this one as it does the same.

File: core/src/main/java/io/kestra/core/utils/LabelUtils.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import io.kestra.core.models.Label;
+
+import java.util.List;
+import java.util.Map;
+
+public final class LabelUtils {
+    private LabelUtils() {
+        //utility class pattern
+    }
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply core/src/main/java/io/kestra/core/utils/LabelUtils.java:11]: Well, it'll do the same for one less line in a less declarative manner so I really don't see the point here.

File: ui/src/components/layout/Labels.vue (hunk 1 above)
Comment (brian-mulier-p) [reply ui/src/components/layout/Labels.vue]: Made the change and tried it, works :)

**Pull request merged**